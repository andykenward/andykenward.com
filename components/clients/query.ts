import { DocumentNode } from "graphql"
import gql from "graphql-tag"

export const FRAGMENT_CLIENT: DocumentNode = gql`
  fragment FragmentClient on Client {
    id
    title
    href
  }
`
