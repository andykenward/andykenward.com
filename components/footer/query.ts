import { DocumentNode } from "graphql"
import gql from "graphql-tag"

export const QUERY_FOOTER: DocumentNode = gql`
  query Footer {
    footer(where: { id: "cjzjk2d1gvch50887pxsk6yju" }) {
      id
      links {
        ...Link
      }
    }
  }
`
