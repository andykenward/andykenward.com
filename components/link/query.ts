import gql from "graphql-tag"

gql`
  fragment Link on Link {
    id
    title
    url
  }
`
