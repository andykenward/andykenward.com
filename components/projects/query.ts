import { DocumentNode } from "graphql"
import gql from "graphql-tag"

export const QUERY_PROJECTS: DocumentNode = gql`
  query Projects {
    projects {
      id
      slug
      title
    }
  }
`

gql`
  fragment ProjectNavItem on Project {
    id
    clients {
      ...Clients
    }
    slug
    title
    description
    role
    studio {
      id
      title
      url
    }
    tags {
      ...Tags
    }
    urlCaseStudy
    url
  }
`

export const QUERY_PROJECTS_NAV: DocumentNode = gql`
  query ProjectsNav {
    projects {
      ...ProjectNavItem
    }
  }
`
