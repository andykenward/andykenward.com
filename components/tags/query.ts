import { DocumentNode } from "graphql"
import gql from "graphql-tag"

export const FRAGMENT_TAG: DocumentNode = gql`
  fragment FragmentTag on Tag {
    id
    title
    href
  }
`
