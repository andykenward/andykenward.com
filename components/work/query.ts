import { DocumentNode } from "graphql"
import gql from "graphql-tag"

export const QUERY_WORK: DocumentNode = gql`
  query Work {
    projects(orderBy: date_DESC) {
      __typename
      id
      slug
      title
    }
  }
`
