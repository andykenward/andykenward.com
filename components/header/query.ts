import { DocumentNode, FragmentSpreadNode } from "graphql"
import gql from "graphql-tag"

import { FRAGMENT_LINK } from "../link/query"

export const FRAGMENT_ME: FragmentSpreadNode = gql`
  fragment FragmentMe on Me {
    id
    name
    role
    update
    links {
      ...FragmentLink
    }
  }
  ${FRAGMENT_LINK}
`

export const QUERY_ME: DocumentNode = gql`
  query Me {
    me(where: { id: "cjzkx8reiebu10941932j2b3x" }) {
      ...FragmentMe
    }
  }
  ${FRAGMENT_ME}
`
