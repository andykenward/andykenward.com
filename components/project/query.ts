import { DocumentNode } from "graphql"
import gql from "graphql-tag"

import { FRAGMENT_PROJECTNAV } from "../projects/query"

export const QUERY_PROJECTS: DocumentNode = gql`
  query Project($slug: String) {
    project(where: { slug: $slug }) {
      ...FragmentProjectNav
    }
  }
  ${FRAGMENT_PROJECTNAV}
`
