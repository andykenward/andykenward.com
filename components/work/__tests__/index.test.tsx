import { useQuery } from "@apollo/react-hooks"
import { wait } from "@apollo/react-testing"
import { RenderResult } from "@testing-library/react"
import { act } from "react-dom/test-utils"

import { Work } from ".."
import { getMock, getMocksError, renderWithApp } from "../../../lib/test-utils"
import { QUERY_WORK } from "../query"
import { WORK_RESULT_DATA } from "./__fixtures__"

describe("Work", () => {
  it("handles work items query", async () => {
    const MOCKS = getMock(QUERY_WORK, WORK_RESULT_DATA)
    const Component = () => {
      const { data, loading } = useQuery(QUERY_WORK)
      if (!loading) {
        expect(data).toEqual(WORK_RESULT_DATA)
      }
      return null
    }
    act(() => {
      renderWithApp(<Component />, MOCKS)
    })
  })

  it("renders loading state", async () => {
    const MOCKS = getMock(QUERY_WORK, WORK_RESULT_DATA)

    await act(async () => {
      const {
        container: { firstChild },
      } = renderWithApp(<Work />, MOCKS)

      expect(firstChild).not.toBeNull()
      expect(firstChild).toMatchSnapshot()
    })
  })

  it("renders error state", async () => {
    const MOCKS = getMocksError(QUERY_WORK)

    await act(async () => {
      const {
        container: { firstChild },
      } = renderWithApp(<Work />, MOCKS)

      await wait(0)

      expect(firstChild).not.toBeNull()
      expect(firstChild).toMatchSnapshot()
    })
  })

  it.todo("handles null or undefined data response")

  describe("render", () => {
    it.todo("renders correct length of items")
    it.todo("shows correct title")
    it.todo("shows correct tags")
    it.todo("shows correct clients")
    it.todo("shows correct studio")
    it.todo("has correct href")
    // use srcset
    // retina images
    // media breakpoints
    // TODO: image api tests from GraphCMS, static export of the images?
    // TODO: look into checking correct image is loaded from srcset?
    // https://graphcms.com/docs/assets/transformations
    // https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Responsive_images
    it.todo("shows correct responsive images")

    it.todo("has :hover css")
    it.todo("has :visited css")
    it.todo("has correct tabindex")
  })
})
