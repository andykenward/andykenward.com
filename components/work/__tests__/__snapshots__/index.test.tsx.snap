// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Work renders error state 1`] = `
<div>
  error
</div>
`;

exports[`Work renders loading state 1`] = `
<div>
  loading
</div>
`;
